Use Blender directly instead of blenderproc

azazdeaz opened the pull request:
Originally, i used blenderproc because it seemed like a convenient way to install and call Blender. Unfortunately it doesn't work with new versions of Blender and has some other issues so in this PR:
 - use Blender directly instead of blenderproc (this requires a `blender` executable on the PATH but we can change that. Let me know what works best at the production environment)
 - minimizes the use of blender specific scripts, so the generation part is not that separated from the rest of the app
 - optimizes the textures in the .gbl exports. this is just a first step, there is a lot more thing we can optimize to make the models smaller
 
 I leave this as a draft until we have a chance to discuss it.

tomonorman commented on roboprop_client/load_blenderkit.py:
```diff
@@ -92,22 +90,19 @@ def add_demo_world(model_path: Path):
-    # Start a clean Blender proceess
-    bproc.init()
-
+def load_blenerkit_model(asset_base_id: str, output_path: str, model_name: str = None):
```
`blenderkit_model` I think (typo)

azazdeaz replied:
Thank you for catching this!

tomonorman commented on roboprop_client/load_blenderkit.py:
```diff
@@ -92,22 +90,19 @@ def add_demo_world(model_path: Path):
+    if not model_name:
+        model_name = meta["name"]
 
     blend_file = load_model_from_blenderkit(meta)
```
same here

tomonorman reviewed (approved):
Theres a typo, but I think good. 
No issue with blender having to be available (as long as can be run completely headless (i.e doesnt need a display) which from my test looks like so.

But perhaps add it to the README as a prerequisite?

On a separate note, how do you feel about using `_` and `__` prefixes to internal/private functions, and having all functions on the same level rather than nesting them? I think a matter of personal preferences, but I am starting to see functions nested within functions, within functions

azazdeaz commented:
Thank you for the review Tomo! I fixed the typos and added a note about the Blender dependency to the wiki: https://github.com/art-e-fact/RoboProp/wiki/installation#install-blender-dependency Let me know if that is not the right place for it.

I don't think i ever worked on large python projects, so if you have any request about the code style, i'd be happy to implement them and stick to them in the future.  :+1:

tomonorman reviewed (approved):
Looks good, thankyou!

As for code styling, we can discuss another time. Thanks!

tomonorman commented:
I think good to merge, optimization issue (rerunning logic) will be based on this branch
